File: tests/job_task_promise_test.py
#!/usr/bin/env python
import angle
#
#
from parser_test_helper import *


class JobTest(ParserBaseTest):

		def test_simple(self):
			context.use_tree = False
			assert_equals(parse('do print 3'), 3) # the program waits for the job to finish

		def test_invariance(self):
			skip()
			assert_result_is('thread{print 3}', 'go print 3')

		def test_complex(self):
				skip()
				xs=parse("xs=for i in 1 to 10: go print i")
				# assert_that("elements in xs are 1 to 10, but order different")

File: angle.js/test/job.task_promise_test.js
#!/usr/bin/env python;
// var angle = require('angle');
//;
//;



class JobTest extends (ParserBaseTest) {

		test_simple(){
			context.use_tree = false;
			assert_equals(parse(`do print 3`), 3) // the program waits for the job to finish
		}
		test_invariance(){
			skip();
			assert_result_is(`thread{print 3}`, 'go print 3');
		}
		test_complex(){
				skip();
				xs=parse(`xs=for i in 1 to 10: go print i`);
				// assert_that("elements in xs are 1 to 10, but order different")

		}
}
